lib/std/os/windows.zig

```diff
@@ -564,6 +564,10 @@ pub fn WriteFile(
                 .NOT_ENOUGH_QUOTA => return error.SystemResources,
                 .IO_PENDING => unreachable,
                 .BROKEN_PIPE => return error.BrokenPipe,
+                // https://github.com/ziglang/zig/issues/4196
+                // Prevents infinite loop when stderr/stdout is not available.
+                // TODO Should implement a NoOpOutStream but this is not possible for now because we cannot coerce or cast to File.OutStream
+                .INVALID_HANDLE => return error.Unexpected,
```

Comment:
Thanks for the fix. Note that "Unexpected" is meant to be for error codes that we never expect to see. But the whole point of this fix is that we expect to see INVALID_HANDLE when there is no stdout/stderr available.

So we need to take a different approach here. Our options are:
 * Use an existing error code such as `error.BrokenPipe`. This would make sense to me since stderr/stdout not being available is exactly the situation of a broken pipe.
 * Invent a new error code for this situation.
 * Ignore the error; treat it as writing to a null device.
 * Declare that the code path is unreachable. This would require the caller to make sure they are using a valid handle before calling WriteFile. In this case we would need to go back to getStdOut() and getStdErr() returning a possible error.

I think my favored approach is to use `error.BrokenPipe`. `std.debug.warn` already will ignore this error, so I think this will have the desired behavior. And for code that uses `try` with a stdout/stderr file, it will get a `error.BrokenPipe` as expected.

I would even go so far as to try this:

```zig
if (std.io.getStdOut() == handle) return error.BrokenPipe;
if (std.io.getStdErr() == handle) return error.BrokenPipe;
unreachable;
```

This would assert that the only 2 handles we would return error.BrokenPipe for would be stderr and stdout. Any other INVALID_HANDLE error would cause unreachable, which matches EBADF from posix code:

https://github.com/ziglang/zig/blob/4b1a8464715a9b2af1d9024694282883f3346482/lib/std/os.zig#L111

Reply:
> I think my favored approach is to use `error.BrokenPipe`. `std.debug.warn` already will ignore this error, so I think this will have the desired behavior. And for code that uses `try` with a stdout/stderr file, it will get a `error.BrokenPipe` as expected.

An common reaction to a `error.BrokenPipe` is to `close()` the pipe. I think we need a new error code to differentiate the cases.


Also note that something similar can happen on POSIX systems if someone closes fd 2.

Reply:
We could have a `error.BrokenStdStream` or `error.InvalidStdStream`, but on Windows closing with an invalid handle won't be an issue, and on POSIX if it happens when the underlying stream has been closed then calling `close()` shouldn't be an issue.

But may be you have another use cas in mind.

Reply:
> We could have a `error.BrokenStdStream` or `error.InvalidStdStream`, but on Windows closing with an invalid handle won't be an issue, and on POSIX if it happens when the underlying stream has been closed then calling `close()` shouldn't be an issue.

[`EBADF` is currently marked as `unreachable`](https://github.com/ziglang/zig/blob/49f3defe5bb85801d333f3871d36ce8db7862e54/lib/std/os.zig#L111)
Which means that closing a non-existent file descriptor is undefined behaviour (nasal demons) in release-fast mode.

Reply:
My  bad, I forgot that unreachable is UB in release-fast.

Reply:
Sorry I messed up my git command commits, they should be squashed together.